File: 算法比赛/LeetCode周赛/第97场双周赛/2.py
from bisect import bisect_left, bisect_right, insort_left, insort_right, insort, bisect
from math import ceil, floor, pow, gcd, sqrt, log10, fabs, fmod, factorial, inf, pi, e
from heapq import heapify, heapreplace, heappush, heappop, heappushpop, nlargest, nsmallest
from typing import List, Tuple, Optional, Dict, Set
from collections import defaultdict, Counter, deque, OrderedDict, namedtuple
from itertools import permutations, combinations, combinations_with_replacement, accumulate, count, groupby
from queue import PriorityQueue, Queue, LifoQueue
from functools import lru_cache, reduce

MOD = int(1e9 + 7)
INF = int(1e20)
INFMIN = float('-inf')
INFMAX = float('inf')

# -*- coding: utf-8 -*-
# @Author  : wheat
# @Time    : 2023/02/04 22:29

class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        banned.sort()
        sum = 0
        res = 0
        if len(banned) == n:
            return res
        if maxSum == 1 and 1 in banned:
            return res
        for i in range(1, n + 1):
            if i not in banned:
                sum += i
                if sum <= maxSum:
                    res += 1
                else:
                    return res
        return res

File: Algorithm_Competition/LeetCode/98/2.go
package _8

import (
	"sort"
)

/*
   -*- coding: utf-8 -*-
   @Author  : wheat
   @Time    : 2023/02/19 11:53
*/

func minimizeSum(nums []int) (ans int) {
	sort.Ints(nums)
	n := len(nums)
	ans = nums[n-2] - nums[1]
	ans = min(ans, nums[n-1]-nums[2])
	ans = min(ans, nums[n-3]-nums[0])
	return
}
